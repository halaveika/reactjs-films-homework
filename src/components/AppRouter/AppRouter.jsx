import React from 'react';
import { Routes, Route, Navigate } from 'react-router-dom';
import routes from '../../routes';

const AppRouter = () => (
  <Routes>
    {routes.map((route) => (
      <Route
        path={route.path}
        element={route.element}
        key={route.path}
      />
    ))}
    <Route path="*" element={<Navigate replace to="/" />} />
  </Routes>
);

export default AppRouter;
