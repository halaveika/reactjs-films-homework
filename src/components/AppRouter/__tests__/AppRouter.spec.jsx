import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { act } from 'react-test-renderer';
import AppRouter from '..';

afterEach(() => {
  jest.clearAllMocks();
});

describe('test AppRouter component', () => {
  it('Matches snapshot ShallowRenderer', () => {
    const renderer = new ShallowRenderer();
    let result;
    act(() => {
      result = renderer.render(
        <AppRouter />,
      );
    });

    expect(result).toMatchSnapshot();
  });
});
