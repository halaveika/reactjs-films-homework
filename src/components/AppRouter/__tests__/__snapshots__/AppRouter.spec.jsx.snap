// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AppRouter component Matches snapshot ShallowRenderer 1`] = `
<Routes>
  <Route
    element={<Memo(Connect(MovieList)) />}
    path="/"
  />
  <Route
    element={
      <React.Fragment>
        <Memo(Connect(MovieDetailsPage)) />
        <Memo(Connect(MovieList)) />
      </React.Fragment>
    }
    path="/details"
  />
  <Route
    element={
      <Navigate
        replace={true}
        to="/"
      />
    }
    path="*"
  />
</Routes>
`;
